### PR TITLE
test: enhance test framework

### PR DESCRIPTION
**Pull request by SchrodingerZhu**

### CODE REVIEW

**Comment by SchrodingerZhu** [comment]
@QuarticCat still working in progress.
I will write framework to test generated parser/lexer.

**Comment by QuarticCat** [comment]
You can make it a draft before you complete it.

File: pag-parser/src/lib.rs [hunk 1]
```diff
@@ -281,6 +281,7 @@ pub fn generate_parser(input: &str) -> Result<TokenStream, Error> {
             unreachable_patterns,
             unreachable_code,
             unused_assignments,
+            unused_parens,
```
Comment (QuarticCat) [inline pag-parser/src/lib.rs:284]: What is this for?

Comment (SchrodingerZhu) [reply pag-parser/src/lib.rs:284]: https://github.com/SchrodingerZhu/paguroidea/pull/58#pullrequestreview-1483466121

Comment (QuarticCat) [reply pag-parser/src/lib.rs:284]: should be fixed in #61

**Review by QuarticCat — APPROVED** [review]

**Pull request merged**